### PR TITLE
Fix overriding problems

### PR DESCRIPTION
**Pull request by odersky**
This fixes the missing "ambiguous overload" error of #1240. More work is needed to fix the multiple overriddenSymbols scenario.

Review by @retronym or @DarkDimius

### CODE REVIEW

File: src/dotty/tools/dotc/core/Denotations.scala [hunk 1]
```diff
@@ -282,8 +282,15 @@ object Denotations {
...
             val sym1 = denot1.symbol
             val sym2 = denot2.symbol
-            val sym2Accessible = sym2.isAccessibleFrom(pre)
 
+            if (sym1.exists && sym2.exists &&
+                (sym1 ne sym2) && (sym1.owner eq sym2.owner) &&
+                !sym1.is(Bridge) && !sym2.is(Bridge))
+              // double definition of two methods with same signature in one class;
+              // don't merge them.
+              return NoDenotation
```
Comment (retronym) [inline src/dotty/tools/dotc/core/Denotations.scala]: Can we get here with private `sym1` or `sym2`? That's another case that prevents overriding.

Comment (retronym) [inline src/dotty/tools/dotc/core/Denotations.scala]: Also, (after looking at your next commit), can we get here with constructors? These also cannot be invoved in overriding.

Comment (odersky) [inline src/dotty/tools/dotc/core/Denotations.scala]: In both cases, the existing resolution will pick the right symbol. So these should not be a problem.

**Comment by odersky** [comment]
Going to merge by tomorrow unless there are further comments

**Comment by odersky** [comment]
Added one important commit and rebased to master.

**Comment by smarter** [comment]
> Interestingly, the failure only triggers when -Ycheck options are *disabled*

Any idea why?

**Comment by DarkDimius** [comment]
@odersky I didn't yet have a chance  to see if this fix is sufficient. I'm fine with meging it now if it's needed in the mainline.

**Pull request merged**